engine/api/workflow/as_code.go

```diff
@@ -0,0 +1,189 @@
+		return nil, sdk.WrapError(err, "cannot pull workflow")
+	}
+
+	// Create VCS Operation
+	ope := sdk.Operation{
+		VCSServer:          app.VCSServer,
+		RepoFullName:       app.RepositoryFullname,
+		URL:                "",
+		RepositoryStrategy: app.RepositoryStrategy,
+		Setup: sdk.OperationSetup{
+			Push: sdk.OperationPush{
+				FromBranch: fmt.Sprintf("cdsAsCode-%d", time.Now().Unix()),
+				Message:    "",
+			},
+		},
+	}
+
+	if app.RepositoryStrategy.ConnectionType == "ssh" {
+		ope.URL = repo.SSHCloneURL
+	} else {
+		ope.URL = repo.HTTPCloneURL
+	}
+
+	if wf.FromRepository == "" {
+		ope.Setup.Push.Message = fmt.Sprintf("feat: Enable workflow as code [@%s]", u.Username)
+	} else {
+ 		ope.Setup.Push.Message = fmt.Sprintf("chore: Update workflow [@%s]", u.Username)
+	}
+
+	if err := PostRepositoryOperation(ctx, db, store, *proj, &ope, buf); err != nil {
+		return nil, sdk.WrapError(err, "unable to post repository operation")
+	}
+
+	store.SetWithTTL(cache.Key(CacheOperationKey, ope.UUID), ope, 300)
+
+	return &ope, nil
+}
+
+// UpdateWorkflowAsCodeResult pulls repositories operation and the create pullrequest + update workflow
+func UpdateWorkflowAsCodeResult(ctx context.Context, db *gorp.DbMap, store cache.Store, p *sdk.Project, ope *sdk.Operation, wf *sdk.Workflow, u *sdk.User) {
+	counter := 0
+	for {
+		counter++
+		if err := GetRepositoryOperation(ctx, db, ope); err != nil {
+			log.Error("unable to get repository operation %s: %v", ope.UUID, err)
+			continue
+		}
+		if ope.Status == sdk.OperationStatusError || ope.Status == sdk.OperationStatusDone {
+			store.SetWithTTL(cache.Key(CacheOperationKey, ope.UUID), ope, 300)
+		}
+
+		if ope.Status == sdk.OperationStatusDone {
+			app := wf.Applications[wf.WorkflowData.Node.Context.ApplicationID]
+			vcsServer := repositoriesmanager.GetProjectVCSServer(p, app.VCSServer)
+			if vcsServer == nil {
+				log.Error("postWorkflowAsCodeHandler> No vcsServer found")
+				return
+			}
+			client, errclient := repositoriesmanager.AuthorizedClient(ctx, db, store, vcsServer)
+			if errclient != nil {
+				log.Error("postWorkflowAsCodeHandler> unable to create repositories manager client: %v", errclient)
+				return
+			}
+			request := sdk.VCSPullRequest{
+				Title: ope.Setup.Push.Message,
+				Head: sdk.VCSPushEvent{
+					Branch: sdk.VCSBranch{
+						DisplayID: ope.Setup.Push.FromBranch,
+					},
+					Repo: app.RepositoryFullname,
+				},
+				Base: sdk.VCSPushEvent{
+					Branch: sdk.VCSBranch{
+						DisplayID: ope.Setup.Push.ToBranch,
+					},
+					Repo: app.RepositoryFullname,
+				},
+			}
+			pr, err := client.PullRequestCreate(ctx, app.RepositoryFullname, request)
+			if err != nil {
+				log.Error("postWorkflowAsCodeHandler> unable to create pull request")
+				return
+			}
+			ope.Setup.Push.PRLink = pr.URL
+			wf.FromRepository = ope.URL
```

Comment:
at this point, until the pull request has been merged, what will be the behavior when we start a new workflow ?

Reply:
for now the workflow become as code.